pandas/core/arrays/arrow/dtype.py

```diff
@@ -91,12 +140,9 @@ def _get_common_dtype(self, dtypes: list[DtypeObj]) -> DtypeObj | None:
             ]
         )
         if not isinstance(new_dtype, np.dtype):
-            # If we ever support e.g. Masked[DatetimeArray] then this will change
-            return None
-        try:
-            return type(self).from_numpy_dtype(new_dtype)
-        except (KeyError, NotImplementedError):
             return None
+        pa_dtype = pa.from_numpy_dtype(new_dtype)
```

Comment:
This call can actually raise an error (eg if the numpy dtype is object dtype), so we should maybe keep the NotImplementedError catching from before